plugin/trino-hive/src/test/java/io/trino/plugin/hive/parquet/AbstractTestParquetReader.java

```diff
@@ -221,6 +221,60 @@ public void testCustomSchemaArrayOfStructs()
                 values, values, "self", new ArrayType(structType), Optional.of(customSchemaArrayOfStructs));
     }
 
+    @Test
+    public void testCustomSchemaArrayOfStructsIncludingStructs()
+            throws Exception
+    {
+        MessageType customSchema = parseMessageType("message CustomSchema {" +
+                "optional group self (LIST) {\n" +
```

Comment:
> The added test fails due to null-values handling (an empty list is read when `null` is expected). Am I testing it wrong or it did capture a tricky edge-case?

With `optional group self` instead of `required ...` the null-values test passes. The required field can never be null, so my intuition is that default value (empty list) was returned because of it when null-values were added explicitly by the test routine.

Reply:
I don't think I fully understand the problem here (I don't fully understand the Parquet schema as well so don't worry).
A required field will never return null simply because it is impossible to encode it. The list will be created while reading the file without any data read.

Reply:
This is not really a problem, just extra unit-test that checks nested arrays work properly (such test case didn't exist before or I didn't find it).